src/test/java/io/vertx/kafka/client/tests/AdminClientTest.java

```diff
@@ -659,6 +660,59 @@ public void testListOffsetsNoTopic(TestContext ctx) {
     adminClient.listOffsets(topicPartitionBeginOffsets, ctx.asyncAssertFailure());
   }
 
+  @Test
+  public void testCreateNewPartitionInTopic(TestContext ctx) {
```

Comment:
we would need another test specifying the new assignments and verify that are fullfilled as requested.